### PR TITLE
api: Export Connection.Ping again

### PR DESCRIPTION
**Pull request by mjs**
... and implement a unit test (it was never tested before).

https://github.com/juju/juju/pull/6400 removed Ping but this was being used by some external projects.

### QA

Full unit test suite run and examined behaviour of a HA failover to ensure no regressions.

### CODE REVIEW

File: api/interface.go [hunk 1]
```diff
@@ -178,6 +178,12 @@ type Connection interface {
 	// All the rest are strange and questionable and deserve extra attention
 	// and/or discussion.
 
+	// Something-or-other expects Ping to exist, and *maybe* the heartbeat
```
Comment (wallyworld) [inline api/interface.go]: Can we just say that Ping() is deprecated and to use IsBroken() instead.

Comment (mjs) [inline api/interface.go]: I will, in the next PR where IsBroken gets added.

**Comment by mjs** [comment]
$$merge$$

**Comment by mjs** [comment]
$$merge$$

**Pull request merged**